### ISSUE TITLE
Implementation error in the MergeMertens HDR algorithm

### ISSUE THREAD

**Issue by cedrou** [Itseez/opencv#5807]
OpenCV 3.0

**tl;dr;**
In the `MergeMertensImpl::process` method, the implementation of the formula to compute the well-exposedness map is false and is missing a call to `cv::exp` to get a gaussian curve.

A fix has been implemented, and a pull request is ongoing. 

**long version**
The class `MergeMertensImpl` in the module opencv_photo, implements an exposure fusion algorithm to create well exposed images from a bracketed exposure sequence of photos (view the original [article](http://web4.cs.ucl.ac.uk/staff/j.kautz/publications/exposure_fusion.pdf)). The algorithm computes 3 weight maps used to define the best parts of the input images. One of these maps is used to keep the areas that are well exposed (not underexposed, nor overexposed) and is built using a Gauss curve:
```
   (   (i-0.5)² )
exp( - -------- ), s=0.2 in this implementation
   (     2s²    )
```
In the implemented of this formula in MergeMertensImpl, the call to the exponential function is missing, and therefore the weight map contains negative values which render it unusable. I guess this is why this weight map is not used by default (`wexp = 0`).

Correcting this problem will allow us to set the default value of the well-exposedness weight factor in the constructor to `1.0`, as recommanded in the original article.
